catroid/src/androidTest/java/org/catrobat/catroid/test/devices/mindstorms/nxt/SensorTests.java

```diff
@@ -23,224 +23,165 @@
 
 package org.catrobat.catroid.test.devices.mindstorms.nxt;
 
-import android.support.test.runner.AndroidJUnit4;
-
-import org.catrobat.catroid.common.bluetooth.ConnectionDataLogger;
-import org.catrobat.catroid.common.bluetooth.models.MindstormsNXTTestModel;
 import org.catrobat.catroid.devices.mindstorms.MindstormsConnection;
-import org.catrobat.catroid.devices.mindstorms.MindstormsConnectionImpl;
+import org.catrobat.catroid.devices.mindstorms.MindstormsException;
+import org.catrobat.catroid.devices.mindstorms.nxt.Command;
 import org.catrobat.catroid.devices.mindstorms.nxt.CommandByte;
-import org.catrobat.catroid.devices.mindstorms.nxt.sensors.NXTI2CUltraSonicSensor;
+import org.catrobat.catroid.devices.mindstorms.nxt.CommandType;
+import org.catrobat.catroid.devices.mindstorms.nxt.NXTException;
 import org.catrobat.catroid.devices.mindstorms.nxt.sensors.NXTLightSensor;
 import org.catrobat.catroid.devices.mindstorms.nxt.sensors.NXTLightSensorActive;
 import org.catrobat.catroid.devices.mindstorms.nxt.sensors.NXTSensor;
 import org.catrobat.catroid.devices.mindstorms.nxt.sensors.NXTSensorMode;
 import org.catrobat.catroid.devices.mindstorms.nxt.sensors.NXTSensorType;
 import org.catrobat.catroid.devices.mindstorms.nxt.sensors.NXTSoundSensor;
 import org.catrobat.catroid.devices.mindstorms.nxt.sensors.NXTTouchSensor;
-import org.junit.After;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
+import org.junit.rules.ExpectedException;
 import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
 
-import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.assertNotNull;
+import java.util.Arrays;
 
-@RunWith(AndroidJUnit4.class)
+import static org.catrobat.catroid.devices.mindstorms.nxt.CommandByte.LS_READ;
+import static org.catrobat.catroid.devices.mindstorms.nxt.CommandType.REPLY_COMMAND;
+import static org.catrobat.catroid.devices.mindstorms.nxt.NXTReply.INSUFFICIENT_REPLY_LENGTH_EXCEPTION_MESSAGE;
+import static org.catrobat.catroid.devices.mindstorms.nxt.NXTReply.INVALID_COMMAND_BYTE_EXCEPTION_MESSAGE;
+import static org.catrobat.catroid.devices.mindstorms.nxt.NXTReply.INVALID_FIRST_BYTE_EXCEPTION_MESSAGE;
+import static org.catrobat.catroid.devices.mindstorms.nxt.NXTReply.NO_ERROR;
+import static org.junit.Assert.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.mockito.internal.verification.VerificationModeFactory.times;
+
+@RunWith(Parameterized.class)
 public class SensorTests {
```

Comment:
since this is already touched anyways, please rename to "SensorTest" (our naming convention is to have all test classes end with "Test")

how is this test different from NXTI2CUltraSonicSensorTest?

Reply:
The I2CSensor does different Stuff on getValue. 
And is difficult to test, because it calls the sendAndReceive of the mocked Connection too often and expects different return Values.. Hence I wrote tests which tests the called functions of  NXTI2CUltraSonicSensor.getValue()